packages/base/base64-image.gts

```diff
@@ -0,0 +1,406 @@
+                {{/if}}
+              {{/if}}
+            </div>
+          {{/if}}
+          <BoxelInput
+            data-test-base64-field
+            @errorMessage={{this.error}}
+            @state={{this.base64InputState}}
+            {{on 'change' this.fileChanged}}
+            type='file'
+          />
+        </div>
+      </FieldContainer>
+      <FieldContainer @label='Alt Text' data-test-field='altText'>
+        <@fields.altText />
+      </FieldContainer>
+      <FieldContainer @label='Size' data-test-field='size'>
+        <@fields.size />
+      </FieldContainer>
+      <FieldContainer @label='Height (px)' data-test-field='height'>
+        <@fields.height />
+      </FieldContainer>
+      <FieldContainer @label='Width (px)' data-test-field='width'>
+        <@fields.width />
+      </FieldContainer>
+    </div>
+
+    <style>
+      .base64-edit {
+        display: grid;
+        gap: var(--boxel-sp-lg);
+      }
+      .preview-wrapper {
+        display: flex;
+        align-items: center;
+        justify-content: center;
+        width: 100%;
+        margin-bottom: var(--boxel-sp);
+      }
+      .preview {
+        border-radius: var(--boxel-form-control-border-radius);
+        background-repeat: no-repeat;
+        background-position: center;
+      }
+      .preview-background {
+        width: 100%;
+        border: 1px solid var(--boxel-form-control-border-color);
+        border-radius: var(--boxel-form-control-border-radius);
+        background-color: white;
+        background-size: 24px 24px;
+        background-position:
+          12px 12px,
+          -12px 0,
+          0 0,
+          0 12px;
+        background-image: linear-gradient(
+            45deg,
+            var(--boxel-300) 25%,
+            transparent 25%
+          ),
+          linear-gradient(-45deg, var(--boxel-300) 25%, transparent 25%),
+          linear-gradient(45deg, transparent 75%, var(--boxel-300) 75%),
+          linear-gradient(-45deg, transparent 75%, var(--boxel-300) 75%);
+      }
+      .height-warning {
+        display: flex;
+        align-items: center;
+        color: var(--boxel-error-200);
+        font: 500 var(--boxel-font-sm);
+        letter-spacing: var(--boxel-lsp);
+      }
+      .height-warning svg {
+        min-width: 20px;
+      }
+      .height-warning span {
+        margin-left: var(--boxel-sp-xxs);
+      }
+    </style>
+  </template>
+}
+
+// this allows multiple radio groups rendered on the page
+// to stay independent of one another.
+let groupNumber = 0;
+class ImageSizeField extends FieldDef {
+  static displayName = 'Image Size';
+  static [primitive]: 'actual' | 'contain' | 'cover';
+  static [useIndexBasedKey]: never;
+  static embedded = class Embedded extends Component<typeof this> {
+    <template>
+      {{@model}}
+    </template>
+  };
+  static edit = class Edit extends Component<typeof this> {
+    <template>
+      <div data-test-radio-group={{@fieldName}}>
```

Comment:
Thoughts on spacing these out? This is flex with `space-between`:

<img width="483" alt="Boxel 2023-12-12 12-39-32" src="https://github.com/cardstack/boxel/assets/43280/dab22029-0e2e-4d4a-b3fd-952bfd050272">

Or `gap: var(--boxel-sp)`:

<img width="483" alt="Boxel 2023-12-12 12-39-02" src="https://github.com/cardstack/boxel/assets/43280/15350779-68f8-41d0-ac54-2a2e2d1c99dc">

Reply:
good idea